src/Binding.lua

```diff
@@ -134,7 +134,7 @@ function BindingInternalApi.join(upstreamBindings)
 				disconnect()
 			end
 
-			disconnects = nil
+			disconnects = nil :: any
```

Comment:
it would be better to make the delcaration of `disconnects` have an explicit type that includes nil (eg `local disconnects: Array<any> | nil = {}`). (Once deferred constraint resolution and some other Luau features land, it will not longer greedily assume the first assignment it sees is the only type possibility for the whole closure.)

Reply:
This leads to issues with indexing the `disconnects` variable since Luau now thinks it could be nil at any point. I think it would be a bit overkill to check `if disconnects then` so maybe we should just stick with casting to any

<img width="762" alt="Screen Shot 2022-06-10 at 11 41 59 AM" src="https://user-images.githubusercontent.com/85896114/173130127-c2258429-e478-4da3-b8d7-0946200bbd4e.png">